server/internal/router/route_map.go

```diff
@@ -94,18 +94,20 @@ func (r *routeMap) deleteEngine(engineID string) {
 }
 
 // findLeastLoadedEngine finds the engine with the least number of loaded models.
-func (r *routeMap) findLeastLoadedEngine() (string, error) {
+func (r *routeMap) findLeastLoadedEngine(ignores map[string]bool) (string, error) {
 	r.mu.Lock()
 	defer r.mu.Unlock()
 
-	if len(r.engines) == 0 {
-		return "", fmt.Errorf("no engine available")
-	}
-
 	// modelsByID is a list of models keyed by engine IDs. This includes engines that don't have any models
 	modelsByID := make(map[string][]model)
 	for id := range r.engines {
-		modelsByID[id] = []model{}
+		if !ignores[id] {
```

Comment:
I might not be following the logic correctly, but shouldn't we have a similar logic at line 115 as well?

Reply:
Nice catch! L115 would be a problem when the ready model fails due to an unschedulable error.